src/entities/currency.test.ts

```diff
@@ -9,10 +9,14 @@ describe('Currency', () => {
 
   describe('#equals', () => {
     it('ether on same chains is ether', () => {
-      expect(Ether.onChain(1).equals(Ether.onChain(1)))
+      expect(
+        Ether.onChain(1, '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2').equals(
+          Ether.onChain(1, '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2')
```

Comment:
what if address is different? they're no longer equivalent... bit odd actually..

Reply:
Yeah, that's why I was suggesting we lift the token address up to `BaseCurrency`.

Reply:
native currency doesn't have an address though, so they are semantically different even though they're the same type. i'm ok with just updating equals to also compare the lowercase addresses